packages/core/entity/FileExplorerURL/test/fileexplorerurl.test.ts

```diff
@@ -350,22 +385,53 @@ describe("FileExplorerURL", () => {
                 filters: expectedFilters.map(({ name, value }) => new FileFilter(name, value)),
                 openFolders: [],
                 sortColumn: new FileSort(AnnotationName.FILE_ID, SortOrder.ASC),
+                collection: {
+                    name: mockCollection.name,
+                    version: mockCollection.version,
+                },
             };
             const encodedUrl = FileExplorerURL.encode(components);
             const encodedUrlWithWhitespace = " " + encodedUrl + " ";
 
             // Act
-            const result = FileExplorerURL.validateEncodedFileExplorerURL(
+            const result = await FileExplorerURL.validateEncodedFileExplorerURL(
                 encodedUrlWithWhitespace,
-                annotations
+                annotations,
+                datasetService
             );
 
             // Assert
             expect(result).to.be.undefined;
         });
 
-        it("Returns error message when not in expected JSON format", () => {
+        it("returns error message when dataset could not be found", async () => {
             // Arrange
+            sinon.stub(datasetService, "getDataset").rejects(new Error("No Dataset found"));
+            const components: FileExplorerURLComponents = {
+                hierarchy: [],
+                filters: [],
+                openFolders: [],
+                collection: {
+                    name: mockCollection.name,
+                    version: mockCollection.version,
+                },
+            };
+            const encodedUrl = FileExplorerURL.encode(components);
+
+            // Act
+            const result = await FileExplorerURL.validateEncodedFileExplorerURL(
+                encodedUrl,
+                [],
+                datasetService
+            );
+
+            // Assert
+            expect(result).to.not.be.empty;
+        });
+
+        it("Returns error message when not in expected JSON format", async () => {
+            // Arrange
+            sinon.stub(datasetService, "getDataset").rejects(new Error("No Dataset found"));
```

Comment:
From the test above, I expect this test to fail _because of this_ as well as whatever is happening with the `encodedUrl` below. This test would be stronger if there was just one condition that is under test/could cause a failure, and/or if the assertion was more specific.

Reply:
Ah hmm true! I'll split this up.